support/k8s/k8s-workload-registrar/mode-crd/controllers/utils.go

```diff
@@ -131,8 +132,14 @@ func equalStringSlice(x, y []string) bool {
 
 func containsString(slice []string, s string) bool {
 	for _, item := range slice {
-		if item == s {
-			return true
+		if regex, err := regexp.Compile(item); err == nil {
+			if regex.MatchString(s) {
+				return true
+			}
+		} else {
+			if item == s {
+				return true
+			}
```

Comment:
- I dont think you need to compile, just use [regexp.MatchString()](https://pkg.go.dev/regexp#MatchString). We dont need to support complex queries here.
- change the function to return `err` and handle the error at the caller.